### PR TITLE
Add regex support to _use_legacy_file_structure()

### LINKED ISSUE #150: Support regex with _use_legacy_file_structure()
E.g. if we have extensions with a running index on repositories that use the old file system, we have to support regex with ` _use_legacy_file_structure()`.

### PR DESCRIPTION
**Pull request by frankenjoe**
Closes #150

### CODE REVIEW

File: audbackend/core/artifactory.py [hunk 1]
```diff
@@ -431,6 +438,7 @@ def _use_legacy_file_structure(
             self,
             *,
             extensions: typing.List[str] = None,
+            regex: bool = False,
```
Comment (hagenw) [inline audbackend/core/artifactory.py:441]: What happens if you mix regex based extensions with non-regex based, e.g. `extensions=['\d+.tar.gz', 'tar.gz']`?

Comment (hagenw) [reply audbackend/core/artifactory.py:441]: I'm asking as it seems a little bit critical, that we can provide a list with extensions, some using a regex pattern others not, but we can only provide a single `regex=True`.

Comment (frankenjoe) [reply audbackend/core/artifactory.py:441]: You are not supposed to mix. Either you provide all as regex or none.

File: tests/test_filesystem.py [hunk 2]
```diff
@@ -61,24 +61,61 @@ def test_get_file_interrupt(tmpdir, bad_file_system, backend):
...
+            'file.0.tar/1.0.0/file.0.tar-1.0.0.gz',
+        ),
+        (
+            '/file.0.tar.gz', '1.0.0', [r'\d+.tar.gz'], True,
+            'file/1.0.0/file-1.0.0.0.tar.gz',
+        ),
+        (
+            '/file.99.tar.gz', '1.0.0', [r'\d+.tar.gz'], True,
+            'file/1.0.0/file-1.0.0.99.tar.gz',
+        ),
```
Comment (hagenw) [inline tests/test_filesystem.py:110]: I would add a test for our actual use case:

```suggestion
        ),
        (
            '/file.prediction.99.tar.gz', '1.0.0', [r'prediction.\d+.tar.gz', r'truth.tar.gz'], True,
            'file/1.0.0/file-1.0.0.prediction.99.tar.gz',
        ),
        (
            '/file.truth.tar.gz', '1.0.0', [r'prediction.\d+.tar.gz', r'truth.tar.gz'], True,
            'file/1.0.0/file-1.0.0.truth.tar.gz',
        ),
```

What still does not work for me is to mix TAR.GZ with 99.TAR.GZ, but maybe I just define the wrong search pattern: `r'(\d.)?tar.gz'`?

**Pull request merged**